01-messaging.md

```diff
@@ -310,12 +310,14 @@ For simplicity of diagnosis, it's often useful to tell a peer that something is
    * [`u16`:`len`]
    * [`len*byte`:`data`]
 
-The 2-byte `len` field indicates the number of bytes in the immediately following field.
+1. type: 1 (`warning`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u16`:`len`]
+   * [`len*byte`:`data`]
 
 #### Requirements
 
-The channel is referred to by `channel_id`, unless `channel_id` is 0 (i.e. all bytes are 0), in which case it refers to all channels.
```

Comment:
I don't see why we should get rid of this? The great thing about having `warning` messages is we can now reserve `error` messages specifically for channel-closing issues. Having a "go away, I don't want to ever talk to you again, close your channels, I hate you" message doesn't seem like a thing we should get rid of just-because.

Reply:
I'm not sure it's necessary anymore, it should be preferable to either mutual close all channels when you want to "go away", or if you lost state settle them after `channel_reestablish` with `option_data_loss_protect` (which will create one error per-channel)?

I believe it was removed because all the places where the spec would recommend to close all channels were changed to only send a warning, doesn't it?

Reply:
> I believe it was removed because all the places where the spec would recommend to close all channels were changed to only send a warning, doesn't it?

I don't think this should be the criteria for removing useful things in the protocol. eg I believe we currently send an `null-error` for "feature flags mismatch, we cannot connect to this peer", which seems like the right thing to do. If our counterparty upgrades to a new version which we cannot talk to at all, we should force-close and move on with our lives. `null-error` lets us do that, otherwise we're in a reconnect loop.

Reply:
> we currently send an null-error for "feature flags mismatch, we cannot connect to this peer"

IMHO this is something that will trigger many channel force-close that in practice are really unnecessary...
I believe a better behavior in that case is "send a warning, disconnect and don't initiate reconnection yourself".

We've had this kind of issue with `eclair-mobile`, which doesn't support `option_static_remotekey`, when a new lnd version made it `mandatory` by default. Users connecting their wallet to their own node couldn't connect after an upgrade. But when they noticed this and understood the issue, they simply reconfigured their node to make this feature `optional` and restarted, and everything was back to normal. With the behavior you suggest, they would have lost all their channels, and I think we would have gotten a lot of heat for it in support tickets!

In the general case though, I agree with you that we may want to keep this potential "close all channels" message (there's no strong need to remove it from the spec), let's see what others think.

Reply:
> I believe a better behavior in that case is "send a warning, disconnect and don't initiate reconnection yourself".

Eh, maybe some day we'll change it, but for now we're going to keep it because its not clear what to set all the various timeout constants to.

In any case, its just an example, and the broader point is more that if we are closing all our channels, we should be able to tell our peers that! One of the biggest annoyances with some nodes ignoring `error` messages entirely today is that nodes never learn a channel has been closed until it hits the chain, which may take quite some time depending on feerates.

Reply:
At one stage in c-lightning we sent this error in return for bad gossip messages.  This was, um, dumb.  So for the longest time we have ignored all-channel errors (well, treated them as a warning), and I doubt anyone else did "close all" either.

Reply:
We do :). And just because folks don't currently do it doesn't mean we can't in the future.

Reply:
I think this is the last thing we need to resolve, I'm still highly unconvinced here - yes, some of the historical places all-0 channel ids have been used was nonsense, but the idea that a peer will have a channel with me and I don't want to talk to them anymore and I'm not able to tell them that seems really wrong? Implementing all the logic to stay connected to them, parse their messages for channel IDs, and then fail *those* channels seems substantially complicated, and probably not relevant in cases where you just want a peer to "go away". We can change how we communicate this if backwards compatibility is the concern, but it just seems very strange to me to not be able to communicate this at all (when we currently can).

Reply:
If some implementations find it useful to keep this option, I think we should keep it.

Reply:
Any update on this? It'd be nice to land this PR.